tools/sctrace/README.md

```diff
@@ -0,0 +1,357 @@
+```
+
+## Output
+
+sctrace provides colored output to distinguish between supported and unsupported system calls:
+
+- **Supported calls**: Normal output (or hidden in quiet mode)
+- **Unsupported calls**: Highlighted in red with "unsupported" message
+
+### Example Output
+
+```
+openat(AT_FDCWD, "/etc/passwd", O_RDONLY) = 3
+read(3, "root:x:0:0:root:/root:/bin/bash\n"..., 4096) = 1234
+close(3) = 0
+chmod("/tmp/test", 0755) (unsupported)
+```
+
+## Project Structure
+
+```
+src/
+├── lib.rs              # Library exports
+├── main.rs             # Main application entry point
+├── parameter.rs        # Command-line argument parsing
+├── scml_parser.rs      # SCML language parser
+├── scml_matcher.rs     # Pattern matching engine
+├── strace_parser.rs    # strace output parser
+└── trace.rs            # Trace iteration (online/offline)
+```
+
+## Library API
+
+sctrace can be used as a library in your own Rust projects. The library provides a high-level API for syscall compliance checking without exposing internal implementation details.
+
+### Main Functions
+
+The library exports two primary functions:
+
+- **`check_program(scml_path, program_path, program_args)`**: Traces a running program in real-time and validates its syscalls against SCML patterns
+
+- **`check_logfile(scml_path, input_path)`**: Analyzes an existing strace log file against SCML patterns
+
+Both functions return `Result<Vec<String>, String>` where:
+- `Ok(Vec<String>)`: A vector of error messages for unsupported or unparsable syscalls (empty vector means all syscalls are supported)
+- `Err(String)`: An error occurred during initialization (e.g., failed to parse SCML file or start tracing)
+
+### Usage Examples
+
+#### Check a running program
+
+```rust
+use sctrace::check_program;
+
+let results = check_program(
+    "patterns.scml",
+    "/bin/ls",
+    vec!["-la"]
+).expect("Failed to check program");
+
+if results.is_empty() {
+    println!("✓ All syscalls are supported");
+} else {
+    println!("✗ Found {} issues:", results.len());
+    for error in results {
+        println!("  - {}", error);
+    }
+}
+```
+
+#### Check a log file
+
+```rust
+use sctrace::check_logfile;
+
+let results = check_logfile(
+    "patterns.scml",
+    "strace.log"
+).expect("Failed to check log file");
+
+for error in results {
+    println!("Issue: {}", error);
+}
+```
+
+For detailed API documentation, run:
+
+```bash
+cargo doc --open
+```
+
+## Dependencies
+
+- `clap`: Command-line argument parsing
+- `regex`: Regular expression support
+- `nom`: Parser combinator library
+- `nix`: Unix system interface for process management
+- `tempfile`: Temporary file handling (dev dependency)
+
+## Testing
+
+Run the test suite:
+
+```bash
+cargo test
+```
+
+Run with verbose output:
+
+```bash
+cargo test -- --nocapture
+```
+
+## Troubleshooting
+
+### strace Version Issues
+
+If you encounter version-related errors, ensure strace 5.15+ is installed:
+
+```bash
+strace --version
```

Comment:
Instead of writing up the troubleshooting instructions, the `sctrace` tool should check the version of `strace` before using it and report an error to the user.